resources/design-document.md

```diff
@@ -148,31 +148,33 @@ String medInfo;
 
 
 ## 6.6 _Get Notification Endpoint_
-* Accepts `GET` requests to `/notifications/:customerId`
-* Scans medication table based on customerId and time, returns all Notifications for that customerId at that time window of 15 minutes before and 15 minutes after that time.
+* Accepts `GET` requests to `/notifications/:time` 
```

Comment:
Passing the time in the URI feels a little strange to me. You may want to consider using a query string parameter instead.

Reply:
My comment was more about how you structure the API endpoint(s). A common approach is to the use (optional) query string parameters for things like search terms or sort order or filter criteria, instead of making those part of the URI. We have a couple examples of using query string parameters in the u5 template project [here](https://github.com/nss-se-cohort-3/u5-ProjectTemplate/blob/main/MusicPlaylistServiceLambda/src/main/java/com/nashss/se/musicplaylistservice/lambda/GetPlaylistSongsLambda.java#L19) and [here](https://github.com/nss-se-cohort-3/u5-ProjectTemplate/blob/main/MusicPlaylistServiceLambda/src/main/java/com/nashss/se/musicplaylistservice/lambda/SearchPlaylistsLambda.java#L24).

For example, a request like `GET /notifications?time=202311031823&window=15` could indicate you want to notifications within a window of 15 minutes before or after the passed in time. (Rather than `GET /notifications/202311031823`.)

You bring up an interesting point about what logic to use to decide the time window to query. The approach we've talked about so far is passing in a time that will be used to figure out which notifications to return. That gives the caller the ability to pick a time in the past or the future.

If you don't want to allow that and/or you want to minimize front end logic, you could consider using the time the endpoint is called, instead of letting the client pass it in. For example, a request could look like `GET /notifications/active` or `GET /notification/current`, and the backend would just get the current time when it does the query and calculate the window from that.